executor/brie.go

```diff
@@ -366,23 +636,48 @@ func handleBRIEError(err error, terror *terror.Error) error {
 	return terror.GenWithStackByArgs(err)
 }
 
-func (e *ShowExec) fetchShowBRIE(kind ast.BRIEKind) error {
-	globalBRIEQueue.tasks.Range(func(key, value interface{}) bool {
-		item := value.(*brieQueueItem)
-		if item.info.kind == kind {
-			item.progress.lock.Lock()
-			defer item.progress.lock.Unlock()
-			current := atomic.LoadInt64(&item.progress.current)
-			e.result.AppendString(0, item.info.storage)
-			e.result.AppendString(1, item.progress.cmd)
-			e.result.AppendFloat64(2, 100.0*float64(current)/float64(item.progress.total))
-			e.result.AppendTime(3, item.info.queueTime)
-			e.result.AppendTime(4, item.info.execTime)
-			e.result.AppendNull(5) // FIXME: fill in finish time after keeping history.
-			e.result.AppendUint64(6, item.info.connID)
+func (e *ShowExec) fetchShowBRIE(ctx context.Context, kind ast.BRIEKind) error {
+	sql := fmt.Sprintf(`SELECT id, data_path, status, progress, queue_time, exec_time, finish_time, conn_id, message
+			FROM mysql.brie_tasks WHERE kind = '%s';`, kind.String())
+	rs, err := e.ctx.(sqlexec.SQLExecutor).Execute(ctx, sql)
+	if err != nil {
+		return err
+	}
+	r := rs[0]
+	defer terror.Call(r.Close)
+	req := r.NewChunk()
+	it := chunk.NewIterator4Chunk(req)
+	for {
+		err = r.Next(ctx, req)
+		if err != nil {
+			return err
 		}
-		return true
-	})
+		if req.NumRows() == 0 {
+			break
+		}
+
+		for row := it.Begin(); row != it.End(); row = it.Next() {
+			e.result.AppendUint64(0, row.GetUint64(0))
+			dataPath, err := base64.StdEncoding.DecodeString(row.GetString(1))
+			if err != nil {
+				logutil.Logger(ctx).Error("failed to decode base64", zap.Error(err))
+				continue
+			}
+			e.result.AppendBytes(1, dataPath)
+			e.result.AppendString(2, row.GetString(2))
+			e.result.AppendFloat32(3, row.GetFloat32(3))
+			e.result.AppendTime(4, row.GetTime(4))
+			e.result.AppendTime(5, row.GetTime(5))
+			e.result.AppendTime(6, row.GetTime(6))
+			e.result.AppendUint64(7, row.GetUint64(7))
+			msg, err := base64.StdEncoding.DecodeString(row.GetString(8))
+			if err != nil {
+				logutil.Logger(ctx).Error("failed to decode base64", zap.Error(err))
```

Comment:
Do we need to log more information to help debug the problem if the error happened